src/hotspot/os/posix/signals_posix.cpp

```diff
@@ -763,10 +805,32 @@ typedef int (*os_sigaction_t)(int, const struct sigaction *, struct sigaction *)
 
 static void SR_handler(int sig, siginfo_t* siginfo, ucontext_t* context);
 
+// Semantically compare two sigaction structures. Return true if they are referring to
+// the same handler, using the same flags.
+static bool compare_handler_info(const struct sigaction* sa,
+                                 const struct sigaction* expected_sa) {
+  address this_handler = get_signal_handler(sa);
+  address expected_handler = get_signal_handler(expected_sa);
+  const int this_flags = get_sanitized_sa_flags(sa);
+  const int expected_flags = get_sanitized_sa_flags(expected_sa);
+  return this_handler == expected_handler &&
+         this_flags == expected_flags;
```

Comment:
Could we use brackets here like:

```c
  return ((this_handler == expected_handler) &&
           (this_flags == expected_flags));
```

Reply:
I can add the inside brackets, but don't see the need for the outside ones, there is nothing to bracket off against.